db-tests/src/test/java/gov/nasa/jpl/aerie/database/PlanCollaborationTests.java

```diff
@@ -2663,4 +2562,155 @@ void cancelCleansUpSuccessfully() throws SQLException {
       }
     }
   }
+
+  @Nested
+  class AnchorMergeTests{
+
+    // Method is taken from AnchorTests
+    private void setAnchor(int anchorId, boolean anchoredToStart, int activityId, int planId) throws SQLException {
+      try(final var statement = connection.createStatement()) {
+        if (anchorId == -1) {
+          statement.execute(
+              """
+                  update activity_directive
+                  set anchor_id = null,
+                      anchored_to_start = %b
+                  where id = %d and plan_id = %d;
+                  """.formatted(anchoredToStart, activityId, planId));
+        } else {
+          statement.execute(
+              """
+                  update activity_directive
+                  set anchor_id = %d,
+                      anchored_to_start = %b
+                  where id = %d and plan_id = %d;
+                  """.formatted(anchorId, anchoredToStart, activityId, planId));
+        }
+      }
+    }
+
+    @Test
+    void cantMergeCycle() throws SQLException{
```

Comment:
Whoa, good edge case. I hadn't considered that "valid" anchors in the source plan can become invalid in the target plan... especially since the merge process allows being selective about which directives are carried over...

Reply:
The question arises - how is this situation made visible to the user, and how does a user remedy the situation? I think what you have here - simply don't allow the commit - is definitely reasonable for this stage. It may be worth thinking through what the "optimal" user experience would be 🤔

Reply:
Yeah, although I don't think this is something that can really be checked in the DB prior to the merge getting committed, since it doesn't have all the information regarding conflict resolution until a user goes to commit.

UX-wise, I think that as long as these errors are actually shown to the user (plus or minus some language tweaking to make it more obvious which activity is part of the cycle), that ought to be sufficient, as they don't lose any work from failing to commit.